pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,296 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"io"
+
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// SandboxManager provides basic operations to create/delete and examine the
+// Sandboxes.
+type SandboxManager interface {
+	// Create creates a sandbox based on the given config, and returns the ID
+	// of the new sandbox.
+	Create(config *SandboxConfig) (string, error)
+	// Delete deletes the sandbox with by its ID. If there are any running
+	// containers in the sandbox, they will be terminated as a side-effect.
+	Delete(id string) (string, error)
+	// List lists existing sandboxes, filtered by the given SandboxFilter.
+	List(filter SandboxFilter) []SandboxListItem
+	// Inspect gets the detailed config and status of the sandbox by ID.
+	Inspect(id string) Sandbox
+}
+
+// SandboxConfig holds all the required and optional fields for creating a
+// sandbox.
+type SandboxConfig struct {
+	// Name is the name of the sandbox.
+	Name string
+	// Hostname is the hostname of the sandbox.
+	Hostname string
+	// DNSOptions the DNS configuration the sandbox.
+	DNSOptions *DNSOptions
+	// PortMappings lists the port mappings for the sandbox.
+	PortMappings []PortMapping
+	// Mounts list the mounts to be added to the sandbox's filesystem.
+	Mounts []Mount
+	// Resources specifies the resource requirements for the sandbox.
+	Resources *Resources
+	// Annotations is an unstructured key value map that may be set by external
+	// tools to store and retrieve arbitrary metadata.
+	Annotations map[string]string
+	// Linux contains configurations specific to Linux containers.
+	Linux *LinuxSandboxConfig
+}
+
+// LinuxSandboxConfig holds platform-specific configuraions for Linux
+// containers.
+type LinuxSandboxConfig struct {
+	// CgroupParent is the parent cgroup of the sandbox.
+	CgroupParent string
+	// NamespaceOptions contains configurations for the sanbox's namespaces.
+	NamespaceOptions NamespaceOptions
+	// SELinux is the SELinux context to be applied to all containers.
+	SELinux *api.SELinuxOptions
```

Comment:
What about AppArmor and Selinux? I think they are set per container.

On Mon, May 16, 2016 at 4:57 PM, Dawn Chen <notifications@github.com> wrote:

> In pkg/kubelet/container/interface.go
> <https://github.com/yujuhong/kubernetes/pull/1#discussion_r63446370>:
>
> > +	// Annotations is an unstructured key value map that may be set by external
> > +	// tools to store and retrieve arbitrary metadata.
> > +	Annotations map[string]string
> > +	// Linux contains configurations specific to Linux containers.
> > +	Linux *LinuxSandboxConfig
> > +}
> > +
> > +// LinuxSandboxConfig holds platform-specific configuraions for Linux
> > +// containers.
> > +type LinuxSandboxConfig struct {
> > +	// CgroupParent is the parent cgroup of the sandbox.
> > +	CgroupParent string
> > +	// NamespaceOptions contains configurations for the sanbox's namespaces.
> > +	NamespaceOptions NamespaceOptions
> > +	// SELinux is the SELinux context to be applied to all containers.
> > +	SELinux *api.SELinuxOptions
>
> Do we have SELinuxOption here or PodSecurityContext here?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/yujuhong/kubernetes/pull/1/files/ddd041c68ca94d95fa591dce703932fe5d577dd0#r63446370>
>